lib/src/Base/Geom/DomainImplementation.cxx

```diff
@@ -63,13 +64,37 @@ Bool DomainImplementation::contains(const Point & ) const
   throw NotYetImplementedException(HERE) << "In DomainImplementation::contains(const Point & point) const";
 }
 
+class DomainImplementationContainsSamplePolicy
+{
+  const Sample & input_;
+  DomainImplementation::BoolCollection & output_;
+  const DomainImplementation & domain_;
+
+public:
+  DomainImplementationContainsSamplePolicy(const Sample & input,
+					   DomainImplementation::BoolCollection & output,
+					   const DomainImplementation & domain)
+    : input_(input)
+    , output_(output)
+    , domain_(domain)
+  {
+    // Nothing to do
+  }
+
+  inline void operator()( const TBBImplementation::BlockedRange<UnsignedInteger> & r ) const
+  {
+    for (UnsignedInteger i = r.begin(); i != r.end(); ++i)
+	output_[i] = domain_.contains(input_[i]);
+  } // operator ()
+};  // class DomainImplementationContainsSamplePolicy
+
 /* Check if the given points are inside of the domain */
 DomainImplementation::BoolCollection DomainImplementation::contains(const Sample & sample) const
 {
   const UnsignedInteger size = sample.getSize();
   BoolCollection result(size, 0);
-  for(UnsignedInteger i = 0; i < size; ++ i)
-    result[i] = contains(sample[i]);
+  const DomainImplementationContainsSamplePolicy policy( sample, result, *this );
+  TBBImplementation::ParallelFor( 0, size, policy);
```

Comment:
this got me thinking that this can crash on other types of domains, for example DomainUnion:
```
import openturns as ot

def _exec1(X):
    return [X[0]]

def _exec2(X):
    return [X[1]]

f1 = ot.PythonFunction(2, 1, _exec1)
f2 = ot.PythonFunction(2, 1, _exec2)
l2 = ot.LevelSet(f2, ot.Greater(), 0.0)
l1 = ot.LevelSet(f1, ot.Greater(), 0.0)
u = ot.DomainUnion([l1, l2])
x = ot.Normal(2).getSample(1000)
y = u.contains(x)
```

So DomainImplementation should have a isParallel method:
- in levelset it should return the function flag
- in union/intersection/complement it should consider the flags of all domains in the list
- Interval/MeshDomain: return True

then here use `TBBImplementation::ParallelForIf(isParallel(), 0, size, policy);`